lib/identity_cache/cache_key_generation.rb

```diff
@@ -7,14 +7,17 @@ def self.schema_to_string(columns)
       columns.sort_by(&:name).map{|c| "#{c.name}:#{c.type}"}.join(',')
     end
 
-    def self.denormalized_schema_hash(klass)
+    def self.denormalized_schema_hash(klass, seen_assocs = Set.new)
       schema_string = schema_to_string(klass.columns)
       klass.send(:all_cached_associations).sort.each do |name, options|
         klass.send(:check_association_scope, name)
         ParentModelExpiration.check_association(options) if options[:embed]
         case options[:embed]
         when true
-          schema_string << ",#{name}:(#{denormalized_schema_hash(options[:association_reflection].klass)})"
+          unless seen_assocs.add?(name)
+              return IdentityCache.memcache_hash(schema_to_string(klass.columns))
```

Comment:
Yeah, because for `seen_assocs.add?` to falsey, the current class must have been added to `seen_assocs` sometime before in a previous call up the call stack. In that call, `unless seen_assocs.add?(name)` wouldn't pass and we would continue execution onto the other elements in `klass.send(:all_cached_associations).sort`.

In other words, there's no point of continuing execution if we've seen an association because by that point, all that model's associations have already been accounted for.